python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -644,7 +644,8 @@ def replicate(self, destSE, sourceSE=''):
         logger.info("Replicating file %s to %s" % (self.lfn, destSE))
         stdout = execute('replicateFile("%s", "%s", "%s")' % (self.lfn, destSE, sourceSE))
         if isinstance(stdout, dict) and stdout.get('OK', False) and self.lfn in stdout.get('Value', {'Successful': {}})['Successful']:
-            self.locations.append(destSE)
+            if destSE not in self.locations:
```

Comment:
Can we make sure we don't try and replicate to their in the first place as this will likely fail in DIRAC and produce a not nice error.
We can catch this higher up and give a nicer error to the user about why we'e not replicating a file.